src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelist.java

```diff
@@ -138,12 +140,15 @@ public final void reset(Collection<? extends Whitelist> delegates) {
                     this.delegates.add(delegate);
                 }
             }
-            for (ProxyWhitelist pw : wrappers.keySet()) {
-                pw.reset();
+            Iterator<ProxyWhitelist> pwIterator = wrappers.keySet().iterator();
```

Comment:
I think this would need to be inside the `synchronized` block.

Reply:
`keySet()` access is synchronized and I was following the documentation:

```
It is imperative that the user manually synchronize on the returned map when traversing any of its collection views via Iterator, Spliterator or Stream:
        Map m = Collections.synchronizedMap(new HashMap());
            ...
        Set s = m.keySet();  // Needn't be in synchronized block
            ...
        synchronized (m) {  // Synchronizing on m, not s!
            Iterator i = s.iterator(); // Must be in synchronized block
            while (i.hasNext())
                foo(i.next());
        }
```

But I guess it does not hurt to add it.